pytorch_lightning/plugins/training_type/ddp_spawn.py

```diff
@@ -208,28 +199,11 @@ def new_process(self, trainer: "pl.Trainer", mp_queue: SimpleQueue) -> None:
         self.barrier()
 
         results = trainer.run_stage()
-
-        # persist info in ddp_spawn
-        self.__transfer_distrib_spawn_state_on_fit_end(trainer, results)
+        outputs = self.__collect_rank_zero_results(trainer, results)
 
         # ensure that spawned processes go through teardown before joining
         trainer._call_teardown_hook()
-
-    def post_dispatch(self, trainer: "pl.Trainer"):
-        # restore main state with best weights
-        best_path = self.mp_queue.get()
-        last_path = self.mp_queue.get()
-        self._results = self.mp_queue.get()
-        # get the `callback_metrics` and set it to the trainer
```

Comment:
Reviewer: these get calls (and corresponding put calls) are no longer needed. The core of this PR is to move the whole tuple of results all at once. This now happens centrally inside the `spawn()` method, aligning both Lite and Lightning.